generator/src/cdn/webpack-customizer.ts

```diff
@@ -102,11 +109,6 @@ export function customizeWebpackConfig(
             })
         );
 
-        // Use hashed module IDs to ease caching support
-        // and avoid the hash-based chunk names being changed
-        // unexpectedly
-        baseConfig.plugins.push(new webpack.HashedModuleIdsPlugin());
-
         // Insert a custom loader to override file and url loaders,
```

Comment:
Since the newly generated webpack config does not use url-loader/file-loader anymore, how is the replacement of the loader with the cdn/webpack-loader.js done? I suspect this would not work when deployed to a cdn.

Reply:
Webpack's 4 `HashedModuleIdsPlugin` corresponds to Webpack's 5 `optimization.moduleIds: 'deterministic'`
https://webpack.js.org/migrate/5/#update-outdated-options
See `HashedModuleIdsPlugin → optimization.moduleIds: 'deterministic'`

And `optimization.moduleIds: 'deterministic'` is default now.
https://webpack.js.org/configuration/optimization/#optimizationmoduleids

That means it's not needed anymore.
BTW, with Webpack 5 it fails with:
```
[webpack-cli] TypeError: webpack.HashedModuleIdsPlugin is not a constructor
    at Object.customizeWebpackConfig (/home/theia-dev/theia-source-code/examples/assembly/cdn/webpack-customizer.js:96:33)
    at module.exports (/home/theia-dev/theia-source-code/examples/assembly/cdn/webpack.config.js:17:26)
    at /home/theia-dev/theia-source-code/node_modules/webpack-cli/lib/webpack-cli.js:1391:43
    at Array.map (<anonymous>)
    at evaluateConfig (/home/theia-dev/theia-source-code/node_modules/webpack-cli/lib/webpack-cli.js:1383:24)
    at /home/theia-dev/theia-source-code/node_modules/webpack-cli/lib/webpack-cli.js:1414:53
    at async Promise.all (index 0)
    at async WebpackCLI.resolveConfig (/home/theia-dev/theia-source-code/node_modules/webpack-cli/lib/webpack-cli.js:1413:38)
    at async WebpackCLI.createCompiler (/home/theia-dev/theia-source-code/node_modules/webpack-cli/lib/webpack-cli.js:1837:22)
    at async WebpackCLI.buildCommand (/home/theia-dev/theia-source-code/node_modules/webpack-cli/lib/webpack-cli.js:1952:20)
    at async Command.<anonymous> (/home/theia-dev/theia-source-code/node_modules/webpack-cli/lib/webpack-cli.js:742:25)
    at async Promise.all (index 1)
    at async Command.<anonymous> (/home/theia-dev/theia-source-code/node_modules/webpack-cli/lib/webpack-cli.js:1289:13)

Error: webpack exited with an unexpected code: 2.
    at ChildProcess.<anonymous> (/home/theia-dev/theia-source-code/dev-packages/application-manager/lib/application-process.js:67:28)
    at ChildProcess.emit (events.js:314:20)
    at maybeClose (internal/child_process.js:1022:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
error Command failed with exit code 1.
```

Reply:
I was not talking about the hashed module id's, but about the insertion of `cdn/webpack-loader.js` We're doing some pattern matching which will never execute. I can't comment on code that has not changed (stupid github)

Reply:
I see. I'm not sure yet how this part actually works.